compiler/src/middle_end/linearize.re

```diff
@@ -52,16 +53,29 @@ let lookup_symbol = (~allocation_type, mod_, mod_decl, name, original_name) => {
     let fresh = gensym(name);
     switch (mod_decl.md_filepath) {
     | Some(filepath) =>
+      let shape =
+        switch (repr) {
+        | ReprFunction(args, rets, Direct(_)) =>
```

Comment:
Curious if/how this branch ever matches? Looking at `middle_end` and `typing`, everything looks like it's `Unknown` at this point in the pipeline?